static/js/libs/channels.js

```diff
@@ -0,0 +1,262 @@
+    format.branch = true;
+  }
+
+  return {
+    track: channelArr[0],
+    risk: channelArr[1],
+    branch: channelArr[2],
+    format: format
+  };
+}
+
+/**
+ * Create a tree from a list of channels
+ * @param {[{track: string, risk: string, branch: string, format: {track: boolean, branch: boolean, format: boolean}}]} channelList
+ * @returns {{
+ *    track: {
+ *      name: string,
+ *      risks: {
+ *        risk: {
+ *          name: string,
+ *          branches: {
+ *            branch: {
+ *              name: string
+ *            }
+ *          }
+ *        }
+ *      }
+ *    }
+ *  }}
+ */
+function createChannelTree(channelList) {
+  const tracks = {};
+
+  channelList.forEach(channel => {
+    if (!tracks[channel.track]) {
+      tracks[channel.track] = {
+        name: channel.track,
+        risks: {}
+      };
+    }
+
+    let level = tracks[channel.track];
+
+    if (!level.risks[channel.risk]) {
+      level.risks[channel.risk] = {
+        name: channel.risk
+      };
+    }
+
+    level = level.risks[channel.risk];
+
+    if (!level.branches) {
+      level.branches = {};
+    }
+
+    level.branches[channel.branch] = {
+      name: channel.branch
+    };
+  });
+
+  return tracks;
+}
+
+/**
+ * Sort a list of strings
+ * The output order will be:
+ *  - hoistValue
+ *  - strings
+ *    - ascending
+ *  - numbers
+ *    - descending
+ * @param {[string]} list
+ * @param {string} hoistValue A value to always appear at the top
+ * @returns {*[]}
+ */
+function sortAlphaNum(list, hoistValue) {
+  let numbers = [];
+  let strings = [];
+  let latest = [];
+  list.forEach(item => {
+    // numbers are defined by any string starting any of the following patterns:
+    //   just a number – 1,2,3,4,
+    //   numbers on the left in a pattern – 2018.3 , 1.1, 1.1.23 ...
+    //   or numbers on the left with strings at the end – 1.1-hotfix
+    if (hoistValue && item === hoistValue) {
+      latest.push(item);
+    } else if (isNaN(parseInt(item.substr(0, 1)))) {
+      strings.push(item);
+    } else {
+      numbers.push(item);
+    }
+  });
+
+  // Ignore case
+  strings.sort(function(a, b) {
+    return a[0].toLowerCase().localeCompare(b[0].toLowerCase());
+  });
+
+  strings = latest.concat(strings);
+
+  // Sort numbers (that are actually strings)
+  numbers.sort((a, b) => {
+    return b[0].localeCompare(a[0], undefined, {
+      numeric: true,
+      sensitivity: "base"
+    });
+  });
+
+  // Join the arrays together again
+
+  return strings.concat(numbers);
+}
+
+/**
+ * Sort channels into the following format:
+ * defaultTrack/stable
+ * defaultTrack/stable/branches
+ * defaultTrack/candidate
+ * defaultTrack/candidate/branches
+ * defaultTrack/beta
+ * defaultTrack/beta/branches
+ * defaultTrack/edge
+ * defaultTrack/edge/branches
+ * track/stable
+ * track/stable/branches
+ * ...
+ *
+ * @param {[string]} channels An array of "risk" / "track/risk" / "track/risk/branch"
+ * @param {{defaultTrack: string}} options
```

Comment:
If there is no other option then `defaultTrack` maybe it just could be a `defaultTrack` param?

Reply:
I wanted to make it generic for future and be more descriptive in the code that calls it. I find:
`sortChannels([], "randomString")` harder to parse than `sortChannels([], { defaultTrack: "randomString" })` when scanning code... I'm open to arguments against though!

Reply:
Not sure if "randomString" would be used in such way directly in code. I would guess in most cases default track would be passed as a variable, so it would be something more like `sortChannels([...], defaultTrack)` which would be more clear.

But, I agree that unnamed option params may be confusing (especially when passing booleans), so I'm not against for sure.

In this case I just wondered if there is any reason to pass it via object. And because it seems like it's used like that consistently in at least 2 functions I'm fine with leaving it as it is.
Code is clear to read, and works, so not worth spending time changing it for little benefit.